addons/zeus/functions/fnc_bi_moduleCurator.sqf

```diff
@@ -0,0 +1,328 @@
+                    getplayeruid player == _ownerVar
+                };
+                case (_isAdmin): {
+                    isserver || servercommandavailable _serverCommand
+                };
+                default {
+                    player == missionnamespace getvariable [_ownerVar,objnull]
+                };
+            };
+            if (_isCurator) then {
+                [true,true] spawn bis_fnc_forceCuratorInterface;
+                ("RscDisplayCurator" call bis_fnc_rscLayer) cuttext ["","black in",1e10];
+            };
+        };
+
+        //--- Check if player is server admin
+        if (_isAdmin) then {
+            _adminVar = _logic getvariable ["adminVar",""];
+            _logic setvariable ["adminVar",nil];
+            if (isserver) then {
+                //--- Host
+                missionnamespace setvariable [_adminVar,player];
+            } else {
+                //--- Client
+                [_logic,_adminVar,_serverCommand] spawn {
+                    scriptname "BIS_fnc_moduleCurator: Admin check";
+
+                    _logic = _this select 0;
+                    _adminVar = _this select 1;
+                    _serverCommand = _this select 2;
+                    while {true} do {
+                        waituntil {sleep 0.1; servercommandavailable _serverCommand};
+                        missionnamespace setvariable [_adminVar,player];
+                        publicvariable _adminVar;
+                        _respawn = player addeventhandler ["respawn",format ["%1 = _this select 0; publicvariable '%1';",_adminVar]];
+
+                        waituntil {sleep 0.1; !servercommandavailable _serverCommand};
+                        missionnamespace setvariable [_adminVar,objnull];
+                        publicvariable _adminVar;
+                        player removeeventhandler ["respawn",_respawn];
+                    };
+                };
+            };
+        };
+
+        [_logic] spawn {
+            _logic = _this select 0;
+            sleep 1;
+            waituntil {alive player};
+
+            //--- Show warning when Zeus key is not assigned
+            if (count (actionkeys "curatorInterface") == 0) then {
+                [
+                    format [
+                        localize "str_a3_cfgvehicles_modulecurator_f_keyNotAssigned",
+                        (["IGUI","WARNING_RGB"] call bis_fnc_displaycolorget) call bis_fnc_colorRGBAtoHTML
+                    ]
+                ] call bis_fnc_guiMessage;
+            };
+
+            //--- Show hint about pinging for players
+            if (
+                isnil {profilenamespace getvariable "bis_fnc_curatorPinged_done"}
+                &&
+                {isTutHintsEnabled}
+                &&
+                {isnull getassignedcuratorlogic player}
+                &&
+                {player in curatoreditableobjects _logic}
+            ) then {
+                sleep 0.5;
```

Comment:
We have our own hinting system, this should be migrated to that instead of the BIS zeus overlay hints.